kpops/components/base_components/kafka_connector.py

```diff
@@ -90,47 +148,27 @@ def connector_config_should_have_component_name(
         app["name"] = component_name
         return KafkaConnectorConfig(**app)
 
+    @computed_field
     @cached_property
-    def helm(self) -> Helm:
-        """Helm object that contains component-specific config such as repo."""
-        helm_repo_config = self.repo_config
-        helm = Helm(self.config.helm_config)
-        helm.add_repo(
-            helm_repo_config.repository_name,
-            helm_repo_config.url,
-            helm_repo_config.repo_auth_flags,
-        )
-        return helm
-
-    @property
-    def _resetter_release_name(self) -> str:
-        suffix = "-clean"
-        return create_helm_release_name(self.full_name + suffix, suffix)
-
-    @property
-    def _resetter_helm_chart(self) -> str:
-        return f"{self.repo_config.repository_name}/kafka-connect-resetter"
-
-    @cached_property
-    def dry_run_handler(self) -> DryRunHandler:
-        helm_diff = HelmDiff(self.config.helm_diff_config)
-        return DryRunHandler(self.helm, helm_diff, self.namespace)
-
-    @property
-    def helm_flags(self) -> HelmFlags:
-        """Return shared flags for Helm commands."""
-        return HelmFlags(
-            **self.repo_config.repo_auth_flags.model_dump(),
-            version=self.version,
-            create_namespace=self.config.create_namespace,
-        )
-
-    @property
-    def template_flags(self) -> HelmTemplateFlags:
-        """Return flags for Helm template command."""
-        return HelmTemplateFlags(
-            **self.helm_flags.model_dump(),
-            api_version=self.config.helm_config.api_version,
+    def _resetter(self) -> KafkaConnectorResetter:
+        kwargs: dict[str, Any] = {}
+        if self.resetter_namespace:
+            kwargs["namespace"] = self.resetter_namespace
+        return KafkaConnectorResetter(
+            config=self.config,
+            handlers=self.handlers,
+            **kwargs,
+            **self.model_dump(
+                exclude={"_resetter", "resetter_values", "resetter_namespace", "app"}
+            ),
+            app=KafkaConnectorResetterValues(
+                connector_type=self._connector_type.value,
+                config=KafkaConnectorResetterConfig(
+                    connector=self.full_name,
+                    brokers=self.config.kafka_brokers,
+                ),
+                **self.resetter_values.model_dump(),
```

Comment:
Will the nameOverride value be in the dump here?
```
name_override=self.full_name + "-clean",
```

Reply:
it's assigned the standard way
1. `full_name` is overridden to include the suffix https://github.com/bakdata/kpops/blob/f9aae5cde7f0c6e4c866a6be9f5cb59fe91a0a78/kpops/components/base_components/kafka_connector.py#L52-L55
3. `HelmApp` sets the `name_override` https://github.com/bakdata/kpops/blob/f9aae5cde7f0c6e4c866a6be9f5cb59fe91a0a78/kpops/components/base_components/helm_app.py#L176-L177

It's verified by the tests, e.g. https://github.com/bakdata/kpops/blob/f9aae5cde7f0c6e4c866a6be9f5cb59fe91a0a78/tests/components/test_kafka_sink_connector.py#L234